tests/unit/test_piece.py

```diff
@@ -0,0 +1,189 @@
+
+
+def test_piece_char():
+    """Ensures proper function when a piece has a valid piece character"""
+    p = Piece('t', 1, Board.empty())
+    assert p.piece_char == 't'
+
+
+def test_negative_piece_pxl_hex():
+    """Ensures proper function when a piece has a negative pixel-hex"""
+    with pytest.raises(PiecePxlHexOutOfBoundsException):
+        Piece('t', -1, Board.empty())
+
+
+def test_large_piece_pxl_hex():
+    """Ensures proper function when a piece has a pixel-hex > 64-bits"""
+    with pytest.raises(PiecePxlHexOutOfBoundsException):
+        Piece('t', 2**64, Board.empty())
+
+
+def test_piece_pxl_hex_immutability():
+    """Ensures proper function when a piece has a piece character length > 1"""
+    p = Piece('t', 1, Board.empty())
+    with pytest.raises(AttributeError):
+        p.piece_pxl_hex = 2
+
+
+def test_piece_pxl_hex():
+    """Ensures proper function when a piece has a valid piece pixel-hex"""
+    p = Piece('t', 1, Board.empty())
+    assert p.piece_pxl_hex == 1
+
+
+def test_empty_current_coords():
+    """Ensures proper function when a piece has empty current_coords"""
+    p = Piece("Test", 1, Board.empty())
+    assert p.current_coords == (0, 0)
+
+
+def test_negative_x_coord():
+    """Ensures proper function when a piece has a negative x coordinate"""
+    with pytest.raises(NegativePieceCoordinatesException):
+        Piece('t', 1, Board.empty(), (-1, 0))
+
+
+def test_negative_y_coord():
+    """Ensures proper function when a piece has a negative y coordinate"""
+    with pytest.raises(NegativePieceCoordinatesException):
+        Piece('t', 1, Board.empty(), (0, -1))
+
+
+def test_current_coords_immutability():
+    """Ensures proper function when a piece has a piece character length > 1"""
+    p = Piece('t', 1, Board.empty())
+    with pytest.raises(AttributeError):
+        p.current_coords = (1, 1)
+
+
+def test_current_coords():
+    """Ensures proper function when a piece has valid current coordinates"""
+    p = Piece('t', 1, Board.empty(), (1, 1))
+    assert p.current_coords == (1, 1)
+
+
+def test_negative_player_controller():
+    """Ensures proper function when a piece has a negative player controller"""
+    with pytest.raises(NegativePiecePlayerNumException):
+        Piece('t', 1, Board.empty(), player_controller=-1)
+
+
+# TODO uncomment this once board updated with current_players
+# def test_non_current_player_controller():
+#     """Ensures proper function when a piece has a non-current
+#     player controller"""
+#     with pytest.raises(NonCurrentPiecePlayerNumException):
+#         Piece('t', 1, Board(1, 1, [""],
+#                             current_players=[0, 1, 2]), player_controller=3)
+
+
+def test_player_controller():
+    """Ensures proper function when a piece has valid player controller"""
+#         Piece('t', 1, Board(1, 1, [""],
+#                             current_players=[0, 1, 2]), player_controller=3)
+    p = Piece('t', 1, Board.empty(), player_controller=2)
```

Comment:
Btw you can do `@pytest.mark.skip(reason=REASON)` to skip running a test or `@pytest.mark.xfail(reason=REASON)` to run the test but expect it to fail (if it succeeds it'll be marked like a failure, if it fails it'll be effectively ignored)